middleware/parse/parse_test.go

```diff
@@ -98,15 +106,44 @@ func TestParseMiddleware(t *testing.T) {
 		assert.Equal(t, http.StatusOK, result.StatusCode)
 	})
 
-	t.Run("JSON Invalid", func(t *testing.T) {
-		request := testutil.NewTestRequest(http.MethodPost, "/parse", bytes.NewBuffer([]byte("{\"unclosed\"")))
-		request.Header().Set("Content-Type", "application/json")
+	t.Run("JSON Parsing Tests", func(t *testing.T) {
+		tests := []struct {
+			name     string
+			body     []byte
+			expected int
+		}{
+			{
+				name:     "JSON Invalid",
+				body:     []byte("{\"unclosed\""),
```

Comment:
```suggestion
				body:     []byte(`{"unclosed"`),
```

Reply:
Next time we've got to make sure we review it in one go 😛